File: codes/perl/008_operator_ternary.pl
use strict;
use warnings;
use JSON;

print("\n# Ternary Operator in Perl\n");

my $CORRECT_ANSWER = "foo";
print("correct answer: \"$CORRECT_ANSWER\"\n");

my $my_answer;

$my_answer = "bar";
print("my answer: \"$my_answer\"\n");
print("is my answer correct: " . (($my_answer eq $CORRECT_ANSWER) ? "correct!" : "wrong!") . "\n");
# is my answer correct: wrong!

$my_answer = "foo";
print("my answer: \"$my_answer\"\n");
print("is my answer correct: " . (($my_answer eq $CORRECT_ANSWER) ? "correct!" : "wrong!") . "\n");
# is my answer correct: correct!

File: codes/php/008_operator_ternary.php
<?php

echo("\n# Ternary Operator in PHP" . "\n");

$CORRECT_ANSWER = 'foo';
echo("correct answer: \"$CORRECT_ANSWER\"" . "\n");

$my_answer = 'bar';
echo("my answer: \"$my_answer\"" . "\n");
echo("is my answer correct: " . (($my_answer == $CORRECT_ANSWER) ? 'correct!' : 'wrong!') . "\n");
# is my answer correct: wrong!

$my_answer = 'foo';
echo("my answer: \"$my_answer\"" . "\n");
echo("is my answer correct: " . (($my_answer == $CORRECT_ANSWER) ? 'correct!' : 'wrong!') . "\n");
# is my answer correct: correct!
